test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandler.ValidateTokenAsyncTests_e2e.cs

```diff
@@ -0,0 +1,415 @@
+        {
+            string token = testTokenCreator.CreateTokenWithMissingAudience();
+            ValidationParameters validationParameters = new ValidationParameters()
+            {
+                ValidAudiences = ["http://Default.Audience.com"],
+                ValidIssuers = ["http://Default.Issuer.com"],
+                IssuerSigningKeys = [KeyingMaterial.JsonWebKeyRsa256SigningCredentials.Key]
+            };
+            CallContext callContext = new CallContext();
+
+            ValidationResult<ValidatedToken> validationResult = await jsonWebTokenHandler.ValidateTokenAsync(token, validationParameters, callContext, default);
+
+            Assert.False(validationResult.IsValid);
+            Assert.Null(validationResult.Result);
+            Assert.NotNull(validationResult.Error);
+            Assert.IsType<AudienceValidationError>(validationResult.Error);
+            Assert.Contains("IDX10206", validationResult.Error.Message);
+            // IDX10206: Unable to validate audience. The 'audiences' parameter is empty.
+            // This message needs to be updated to specifiy that it was not specified in the ValidationParameters?
+        }
+
+        [Fact]
+        public async Task TestTokenWithMissingIssuedAt()
+        {
+            string token = testTokenCreator.CreateTokenWithMissingIssuedAt();
+            ValidationParameters validationParameters = new ValidationParameters()
+            {
+                ValidAudiences = ["http://Default.Audience.com"],
+                ValidIssuers = ["http://Default.Issuer.com"],
+                IssuerSigningKeys = [KeyingMaterial.JsonWebKeyRsa256SigningCredentials.Key]
+            };
+            CallContext callContext = new CallContext();
+
+            ValidationResult<ValidatedToken> validationResult = await jsonWebTokenHandler.ValidateTokenAsync(token, validationParameters, callContext, default);
+
+            // TODO: is it expected that a token with a missing issued at time is valid?
+            Assert.True(validationResult.IsValid);
+            Assert.NotNull(validationResult.Result);
+            Assert.Null(validationResult.Error);
+        }
+
+        [Fact]
+        public async Task TestTokenWithMissingNotBefore()
+        {
+            string token = testTokenCreator.CreateTokenWithMissingNotBefore();
+            ValidationParameters validationParameters = new ValidationParameters()
+            {
+                ValidAudiences = ["http://Default.Audience.com"],
+                ValidIssuers = ["http://Default.Issuer.com"],
+                IssuerSigningKeys = [KeyingMaterial.JsonWebKeyRsa256SigningCredentials.Key]
+            };
+            CallContext callContext = new CallContext();
+
+            ValidationResult<ValidatedToken> validationResult = await jsonWebTokenHandler.ValidateTokenAsync(token, validationParameters, callContext, default);
+
+            // TODO: is it expected that a token with a missing issued at time is valid?
```

Comment:
@iNinja @brentschmaltz :  is it expected that a token with a missing issued at time is valid?

Reply:
@jmprieur iat is optional.